components/youtube_script_injector/renderer/youtube_injector_frame_js_handler.cc

```diff
@@ -0,0 +1,102 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/youtube_script_injector/renderer/youtube_injector_frame_js_handler.h"
+
+#include "base/no_destructor.h"
+#include "base/strings/utf_string_conversions.h"
+#include "content/public/renderer/render_frame.h"
+#include "gin/arguments.h"
+#include "gin/function_template.h"
+#include "third_party/blink/public/mojom/devtools/console_message.mojom.h"
+#include "third_party/blink/public/platform/browser_interface_broker_proxy.h"
+#include "third_party/blink/public/platform/scheduler/web_agent_group_scheduler.h"
+#include "third_party/blink/public/platform/web_string.h"
+#include "third_party/blink/public/web/blink.h"
+#include "third_party/blink/public/web/web_console_message.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace youtube_script_injector {
+
+YouTubeInjectorFrameJSHandler::YouTubeInjectorFrameJSHandler(
+    content::RenderFrame* render_frame)
+    : render_frame_(render_frame) {}
+
+YouTubeInjectorFrameJSHandler::~YouTubeInjectorFrameJSHandler() = default;
+
+bool YouTubeInjectorFrameJSHandler::EnsureConnected() {
+  if (!youtube_injector_.is_bound()) {
+    render_frame_->GetBrowserInterfaceBroker().GetInterface(
+        youtube_injector_.BindNewPipeAndPassReceiver());
+  }
+
+  return youtube_injector_.is_bound();
+}
+
+void YouTubeInjectorFrameJSHandler::AddJavaScriptObjectToFrame(
+    v8::Local<v8::Context> context) {
+  CHECK(render_frame_);
+  CHECK(!context.IsEmpty());
+  v8::Isolate* isolate =
+      render_frame_->GetWebFrame()->GetAgentGroupScheduler()->Isolate();
+
+  v8::Context::Scope context_scope(context);
+  v8::MicrotasksScope microtasks(isolate, context->GetMicrotaskQueue(),
+                                 v8::MicrotasksScope::kDoNotRunMicrotasks);
+  BindFunctionsToObject(isolate, context);
+}
+
+void YouTubeInjectorFrameJSHandler::ResetRemote(
+    content::RenderFrame* render_frame) {
+  render_frame_ = render_frame;
+  youtube_injector_.reset();
+  EnsureConnected();
+}
+
+void YouTubeInjectorFrameJSHandler::BindFunctionsToObject(
+    v8::Isolate* isolate,
+    v8::Local<v8::Context> context) {
+  v8::Local<v8::Object> global = context->Global();
+  v8::Local<v8::Object> brave_obj;
+  v8::Local<v8::Value> brave_value;
+  if (!global->Get(context, gin::StringToV8(isolate, "brave"))
+           .ToLocal(&brave_value) ||
+      !brave_value->IsObject()) {
+    brave_obj = v8::Object::New(isolate);
+    global->Set(context, gin::StringToSymbol(isolate, "brave"), brave_obj)
+        .Check();
+  } else {
+    brave_obj = brave_value->ToObject(context).ToLocalChecked();
+  }
+  BindFunctionToObject(
+      isolate, brave_obj, "nativePipMode",
```

Comment:
per security review this is a major concern, why is this being added and how is it used?